display_list/display_list_builder.h

```diff
@@ -390,7 +391,7 @@ class DisplayListBuilder final : public virtual Dispatcher,
   void onSetBlender(sk_sp<SkBlender> blender);
   void onSetShader(sk_sp<SkShader> shader);
   void onSetImageFilter(sk_sp<SkImageFilter> filter);
-  void onSetColorFilter(sk_sp<SkColorFilter> filter);
+  void onSetColorFilter(const DlColorFilter* filter);
```

Comment:
Is there any reason to not use a shared_ptr here? Even, for example, a shared_ptr<const DlColorFilter>?

Reply:
If we do that, then when the dispatcher wants to dispatch the op from the record in the DL, it has to reinstantiate the object.

Alternately, I suppose I could back off to storing the shared_ptr in the buffer, but then equals and destructor have to visit each node and manually compare or manually destruct. One of my goals was to embed the info directly in the DL buffer so that we could mass compare and mass free the buffers.